src/context.rs

```diff
@@ -485,6 +485,20 @@ impl Context {
 
     /// Stops the IO scheduler.
     pub async fn stop_io(&self) {
+        if let Some(iroh) = self.iroh.write().await.take() {
+            // Close all QUIC connections.
+            let ctx = self.clone();
+
+            // Spawn into a separate task,
+            // because Iroh calls `wait_idle()` internally
+            // and it may take time, especially if the network
+            // has become unavailable.
+            tokio::spawn(async move {
+                if let Err(err) = iroh.close().await {
```

Comment:
maybe put a timeout on this, so it doesn't hang forever, I don't think we do internally